easybuild/tools/module_generator.py

```diff
@@ -213,55 +213,68 @@ def _filter_paths(self, key, paths):
+        filtered_paths = [x for x in paths if x not in added_paths and not added_paths.add(x)]
         if filtered_paths != paths:
             removed_paths = paths if filtered_paths is None else [x for x in paths if x not in filtered_paths]
             print_warning("Suppressed adding the following path(s) to $%s of the module as they were already added: %s",
-                          key, removed_paths,
+                          key, ', '.join(removed_paths),
                           log=self.log)
-            if not filtered_paths:
-                filtered_paths = None
         return filtered_paths
 
     def append_paths(self, key, paths, allow_abs=False, expand_relpaths=True):
         """
         Generate append-path statements for the given list of paths.
 
         :param key: environment variable to append paths to
-        :param paths: list of paths to append
+        :param paths: single or list of paths to append
         :param allow_abs: allow providing of absolute paths
         :param expand_relpaths: expand relative paths into absolute paths (by prefixing install dir)
         """
-        paths = self._filter_paths(key, paths)
-        if paths is None:
-            return ''
         return self.update_paths(key, paths, prepend=False, allow_abs=allow_abs, expand_relpaths=expand_relpaths)
 
     def prepend_paths(self, key, paths, allow_abs=False, expand_relpaths=True):
         """
         Generate prepend-path statements for the given list of paths.
 
         :param key: environment variable to append paths to
-        :param paths: list of paths to append
+        :param paths: single or list of paths to append
         :param allow_abs: allow providing of absolute paths
         :param expand_relpaths: expand relative paths into absolute paths (by prefixing install dir)
         """
-        paths = self._filter_paths(key, paths)
-        if paths is None:
-            return ''
         return self.update_paths(key, paths, prepend=True, allow_abs=allow_abs, expand_relpaths=expand_relpaths)
 
+    def update_paths(self, key, paths, prepend=True, allow_abs=False, expand_relpaths=True):
+        """
+        Generate append/prepend-path statements for the given list of paths.
+
+        :param key: environment variable to append paths to
+        :param paths: single or list of paths to append
+        :param allow_abs: allow providing of absolute paths
+        :param expand_relpaths: expand relative paths into absolute paths (by prefixing install dir)
+        """
+        if isinstance(paths, str):
+            self.log.debug("Wrapping %s into a list before using it for %s", paths, key)
+            paths = [paths]
+        elif not isinstance(paths, list):
+            paths = list(paths)  # coerce any iterator into a list
+
+        if key == 'PYTHONPATH':
+            python_paths = [path for path in paths if re.match(r'lib/python\d+\.\d+/site-packages', path)]
```

Comment:
also consider `lib64/python*/site-packages`?

Reply:
I guess that would be wrong to have both, so yeah:
```suggestion
            python_paths = [path for path in paths if re.match(r'lib(64)?/python\d+\.\d+/site-packages', path)]
```